### ISSUE TITLE
Error accessing Gaiasource on production service using Spark

### ISSUE THREAD

**Issue by stvoutsin** [wfau/gaia-dmp#1085]
**Zeppelin Instance:**

https://dmp.gaia.ac.uk/#/

**Notebook that caused the error (Forrest Classifier):**

https://raw.githubusercontent.com/wfau/aglais-notebooks/main/Public%20Examples/7.%20Good%20astrometric%20solutions%20via%20ML%20Random%20Forest%20classifier_2GQDKZ59J.zpln

**Spark / Zeppelin error message:**

```
Py4JJavaError: An error occurred while calling o155.count.
: org.apache.spark.SparkException: Job aborted due to stage failure: Task 4 in stage 10.0 failed 4 times, most recent failure: Lost task 4.3 in stage 10.0 (TID 32) (worker06 executor 14): java.io.FileNotFoundException: File file:/data/gaia/GDR3/GDR3_GAIASOURCE/part-00004-18ac60bc-42ce-42dd-ab5b-fc0027a56a2c_00004.c000.snappy.parquet does not exist
It is possible the underlying files have been updated. You can explicitly invalidate the cache in Spark by running 'REFRESH TABLE tableName' command in SQL or by recreating the Dataset/DataFrame involved.
```

**Additional Info**

Checking worker06, it looks like something has gone wrong with the Ceph mounts:

`ls -al /data/gaia
`
```
ls: cannot access '/data/gaia/GEDR3_2048': Permission denied
ls: cannot access '/data/gaia/GDR3_2048': Permission denied
total 16
drwxr-xr-x. 6 root root 4096 Oct 13 19:16 .
drwxr-xr-x. 6 root root 4096 Oct 13 19:15 ..
drwxr-xr-x. 2 root root 4096 Oct 13 19:17 GDR3
d?????????? ? ?    ?       ?            ? GDR3_2048
drwxr-xr-x. 2 root root 4096 Oct 13 19:15 GEDR3
d?????????? ? ?    ?       ?            ? GEDR3_2048
```

**Comment by Zarquan** [comment]
Is it just one worker or all of them ?

**Comment by Zarquan** [comment]
In the past we have unmounted and remounted the CephFS shares to fix a similar problem.

https://github.com/wfau/gaia-dmp/blob/a6d01e8938295c9c8b9b7cf7607acc280dc27f24/notes/zrq/20221005-01-cephfs-debug.txt#L159-L194

However, in the previous case we had a functional backup site that we could test the theory on before applying it to the live system.

Try applying this to a single share on a single worker node and see if it solves the problem ?

**Comment by Zarquan** [comment]
Checked with Paul Browne at Cambridge and they don't have any ongoing issues, so this is a new one.

His advice for debugging was to check if we could ping the Ceph monitor nodes from our VMs. The IP addresses for the Ceph monitor nodes will be in the mount details in `/etc/fstab` on each VM.

**Comment by Zarquan** [comment]
I have seen similar in the past where a network outage caused the CephFS mounts to become disconnected, and they don't re-connect afterwards.

**Comment by Zarquan** [comment]
@stvoutsin Any progress on this ?

**Comment by stvoutsin** [comment]
> Is it just one worker or all of them ?

Looks like its an issue with all except one worker (worker01)

**Comment by Zarquan** [comment]
`/user` directories are broken too

**Comment by stvoutsin** [comment]
Pinging the entries in fstab seemed to work ok on all worker nodes, and I've run the unmount / remount suggested above on worker05 and worker06, and this seems to have fixed the issue. I'll run it on the rest now

**Comment by stvoutsin** [comment]
I've run some tests on dmp.gaia.ac.uk all of which work fine. If it works ok for you as well @Zarquan shall we close the issue?

**Comment by Zarquan** [comment]
@stvoutsin What tests did you run ?

**Comment by stvoutsin** [comment]
I ran the Start Here / Database Holdings / Mean Proper Motions & Forrest classifier notebooks. Have you been able to run notebook successfully?

**Comment by Zarquan** [comment]
I ran this test on all the nodes:

```
ssh fedora@live.gaia-dmp.uk
  
    machines=(
        zeppelin
        worker01
        worker02
        worker03
        worker04
        worker05
        worker06
        master01
        )

    for machine in "${machines[@]}"
    do
        echo
        echo "Machine [${machine}]"
        ssh "${machine}" \
            '
            date
            hostname
            echo
            for mountpoint in $(
                cat /etc/fstab | sed -n "/ceph/ p" | cut -d " " -f 2
                )
            do
                echo "Mount [${mountpoint}]"
                du --summarize --human-readable "${mountpoint}"
            done
            '
     done
```

Results indicate that not all the mounts work on all the nodes.
```
Machine [zeppelin]
Mon 23 Jan 15:58:07 UTC 2023
iris-gaia-blue-20221013-zeppelin
....
Mount [/data/wise/ALLWISE]
du: cannot access '/data/wise/ALLWISE': Permission denied
....
Mount [/home/Oor7cheitoiz8Fie]
du: cannot read directory '/home/Oor7cheitoiz8Fie'0	/home/Oor7cheitoiz8Fie
: Permission denied
Mount [/user/Oor7cheitoiz8Fie]
du: cannot access '/user/Oor7cheitoiz8Fie': Permission denied
Mount [/home/DCrake]
du: cannot read directory '/home/DCrake'0	/home/DCrake
: Permission denied
....
Mount [/home/NHambly]
du: cannot access '/home/NHambly': Permission denied
....
Mount [/home/SVoutsinas]
du: cannot access '/home/SVoutsinas': Permission denied
Mount [/user/SVoutsinas]
-- test hangs at this point, needed to use ^C to move to the next test --

Machine [worker01]
Mon 23 Jan 15:58:47 UTC 2023
iris-gaia-blue-20221013-worker01
....

Machine [worker02]
Mon 23 Jan 15:58:50 UTC 2023
iris-gaia-blue-20221013-worker02
....

Machine [worker03]
Mon 23 Jan 15:58:54 UTC 2023
iris-gaia-blue-20221013-worker03
....
Mount [/user/SVoutsinas]
-- test hangs at this point, needed to use ^C to move to the next test --

Machine [worker04]
Mon 23 Jan 15:59:20 UTC 2023
iris-gaia-blue-20221013-worker04
....

Machine [worker05]
Mon 23 Jan 15:59:21 UTC 2023
iris-gaia-blue-20221013-worker05
....

Machine [worker06]
Mon 23 Jan 15:59:26 UTC 2023
iris-gaia-blue-20221013-worker06
....

Machine [master01]
Mon 23 Jan 15:59:30 UTC 2023
iris-gaia-blue-20221013-master01
....
Mount [/data/gaia/GEDR3_2048]
du: cannot access '/data/gaia/GEDR3_2048': Permission denied
....
Mount [/data/wise/ALLWISE]
du: cannot access '/data/wise/ALLWISE': Permission denied
Mount [/data/panstarrs/PS1]
du: cannot access '/data/panstarrs/PS1': Permission denied
....

```

**Comment by Zarquan** [comment]
Not sure _why_ the Hadoop master node needs to see the data mounts, but as long as they are in `/etc/fastab`  we should maintain them.

**Comment by Zarquan** [comment]
This will be resolved by #1092

**Comment by stvoutsin** [comment]
Closing this as GaiaSource is now accessible again on the most recent deploy, although as @Zarquan noted in our telecon there may be a new Ceph-related issue with one of the user's shares / directories
